Backend/app/rider-platform/rider-app/Main/src/API/UI/SearchRetry.hs

```diff
@@ -0,0 +1,154 @@
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE OverloadedLists #-}
+
+module API.UI.SearchRetry
+  ( SearchRetryReq (..),
+    SearchRetryRes (..),
+    DSearchCommon.SearchReqLocation (..),
+    API,
+    handler,
+  )
+where
+
+import qualified Beckn.ACL.Search as TaxiACL
+import Data.Aeson
+import Data.OpenApi hiding (Header)
+import qualified Data.Text as T
+import qualified Domain.Action.UI.Search.Common as DSearchCommon
+-- import Kernel.Types.Version
+
+import Domain.Action.UI.Search.OneWay as OneWay
+import Domain.Action.UI.SearchRetry.OneWayRetry as DOneWayRetry
+import qualified Domain.Types.Person as Person
+import Domain.Types.SearchRequest (SearchRequest)
+import Environment
+import qualified Kernel.External.Slack.Flow as SF
+import Kernel.External.Slack.Types (SlackConfig)
+import Kernel.Prelude
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Common hiding (id)
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Types.SlidingWindowLimiter
+import Kernel.Utils.Common
+import Kernel.Utils.SlidingWindowLimiter
+import Servant hiding (throwError)
+import qualified SharedLogic.CallBPP as CallBPP
+import Storage.CachedQueries.CacheConfig
+import qualified Storage.CachedQueries.Merchant as QMerchant
+import qualified Storage.CachedQueries.RouteInfo as Redis
+import qualified Storage.Queries.Person as Person
+import Tools.Auth
+import qualified Tools.Maps as Maps
+import Tools.Metrics
+
+------ Search Flow --------
+
+type API =
+  "rideSearchRetry"
+    :> TokenAuth
+    :> ReqBody '[JSON] SearchRetryReq
+    :> Capture "parentSearchId" (Id SearchRequest)
+    :> Header "device" Text
+    :> Post '[JSON] SearchRetryRes
+
+handler :: FlowServer API
+handler = search
+
+data SearchRetryRes = SearchRetryRes
+  { searchId :: Id SearchRequest,
+    searchExpiry :: UTCTime,
+    routeInfo :: Maybe Maps.RouteInfo
+  }
+  deriving (Generic, FromJSON, ToJSON, Show, ToSchema)
+
+search :: Id Person.Person -> SearchRetryReq -> Id SearchRequest -> Maybe Text -> FlowHandler SearchRetryRes
+search personId req parentSearchId device = withFlowHandlerAPI . withPersonIdLogTag personId $ do
+  checkSearchRateLimit personId
+  (searchId, searchExpiry, routeInfo) <- searchRetry personId parentSearchId req device
+  return $ SearchRetryRes searchId searchExpiry routeInfo
+
+searchRetry ::
```

Comment:
this function should be moved to `Domain.Action.UI.SearchRetry` module

Reply:
done